scio-repl/src/main/scala/com/spotify/scio/repl/IoCommands.scala

```diff
@@ -0,0 +1,151 @@
+    inputStream(path).asUnsafeCsvReader[T](sep, header).toIterator
+  }
+
+  /** Read from a TSV file on local filesystem, GCS, or HDFS. */
+  def readTsv[T: RowDecoder](path: String,
+                             sep: Char = '\t',
+                             header: Boolean = false): Iterator[T] = {
+    import kantan.csv.ops._
+    implicit val codec = scala.io.Codec.UTF8
+    inputStream(path).asUnsafeCsvReader[T](sep, header).toIterator
+  }
+
+  // =======================================================================
+  // Write operations
+  // =======================================================================
+
+  private def plural[T](data: Seq[T]): String = if (data.size > 1) "s" else ""
+
+  /** Write to an Avro file on local filesystem, GCS, or HDFS. */
+  def writeAvro[T: ClassTag](path: String, data: Seq[T]): Unit = {
+    val cls = ScioUtil.classOf[T]
+    val (writer, schema) = if (classOf[SpecificRecordBase] isAssignableFrom cls) {
+      (new SpecificDatumWriter[T](cls), data.head.asInstanceOf[SpecificRecordBase].getSchema)
+    } else {
+      (new GenericDatumWriter[T](), data.head.asInstanceOf[GenericRecord].getSchema)
+    }
+    val fileWriter = new DataFileWriter[T](writer).create(schema, outputStream(path, BINARY))
+    data.foreach(fileWriter.append)
+    fileWriter.close()
+    println(s"${data.size} record${plural(data)} written to $path")
+  }
+
+  /** Write to a text file on local filesystem, GCS, or HDFS. */
+  def writeText(path: String, data: Seq[String]): Unit = {
+    IOUtils.writeLines(data.asJava, IOUtils.LINE_SEPARATOR, outputStream(path, TEXT))
+    println(s"${data.size} line${plural(data)} written to $path")
+  }
+
+  /** Write to a CSV file on local filesystem, GCS, or HDFS. */
+  def writeCsv[T: RowEncoder](path: String, data: Seq[T],
+                              sep: Char = ',',
+                              header: Seq[String] = Seq.empty): Unit = {
+    import kantan.csv.ops._
+    IOUtils.write(data.asCsv(sep, header), outputStream(path, TEXT))
+  }
+
+  /** Write to a TSV file on local filesystem, GCS, or HDFS. */
+  def writeTsv[T: RowEncoder](path: String, data: Seq[T],
+                              sep: Char = '\t',
+                              header: Seq[String] = Seq.empty): Unit = {
+    import kantan.csv.ops._
+    IOUtils.write(data.asCsv(sep, header), outputStream(path, TEXT))
+  }
+
+  // =======================================================================
+  // Utilities
+  // =======================================================================
+
+  private def inputStream(path: String): InputStream =
+    if (path.startsWith("hdfs://")) {
+      fs.open(new Path(path))
+    } else if (path.startsWith("gs://")) {
+      Channels.newInputStream(gcsUtil.open(GcsPath.fromUri(path)))
+    } else {
+      new FileInputStream(path)
+    }
+
+  private def outputStream(path: String, contentType: String): OutputStream =
+    if (path.startsWith("hdfs://")) {
```

Comment:
I have seen a couple of these - here and there - should we have a function that checks if paths is `hdfs` or `gs`?